### PR TITLE
Feature: improve code and improve golangci-lint config

### LINKED ISSUE #776: improve code and improve golangci-lint config
improve code and improve golangci-lint config

### PR DESCRIPTION
**Pull request by houseme**
**Please provide issue(s) of this PR:**
Fixes #776

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Installation
- [ ] Performance and Scalability
- [ ] Naming
- [X] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by chuntaojun** [comment]
pls fix ci

**Comment by houseme** [comment]
> pls fix ci

已处理

File: apiserver/xdsserverv3/dump.go [hunk 1]
```diff
@@ -38,20 +38,30 @@ func dumpSnapShot(snapshot cache.ResourceSnapshot) []byte {
...
+	if err != nil {
+		return nil
+	}
 	return data
 }
 
 func yamlEncode(any interface{}) []byte {
-	data, _ := json.Marshal(any)
+	data, err := json.Marshal(any)
+	if err != nil {
```
Comment (chuntaojun) [inline apiserver/xdsserverv3/dump.go:55]: 打印下 error 信息

File: apiserver/xdsserverv3/dump.go [hunk 2]
```diff
@@ -38,20 +38,30 @@ func dumpSnapShot(snapshot cache.ResourceSnapshot) []byte {
...
+		return nil
+	}
 	o := make(map[string]interface{})
-	_ = json.Unmarshal(data, &o)
-	data, _ = yaml.Marshal(o)
+	if err = json.Unmarshal(data, &o); err != nil {
```
Comment (chuntaojun) [inline apiserver/xdsserverv3/dump.go:59]: 同上

File: apiserver/xdsserverv3/server.go [hunk 3]
```diff
@@ -955,9 +947,9 @@ func (x *XDSServer) initRegistryInfo() error {
 		return fmt.Errorf("error to init registry info %s", resp.Code)
 	}
 	namespaces := resp.Namespaces
-	// 启动时，获取全量的 namespace 信息，用来推送空配置
-	for _, namespace := range namespaces {
-		x.registryInfo[namespace.Name.Value] = []*ServiceInfo{}
+	// 启动时，获取全量的 n 信息，用来推送空配置
```
Comment (chuntaojun) [inline apiserver/xdsserverv3/server.go:950]: ？ 这里不要替换注释

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: apiserver/xdsserverv3/dump.go (hunk 1 above)
Comment (houseme) [reply apiserver/xdsserverv3/dump.go:55]: done

File: apiserver/xdsserverv3/dump.go (hunk 2 above)
Comment (houseme) [reply apiserver/xdsserverv3/dump.go:59]: done

File: apiserver/xdsserverv3/server.go (hunk 3 above)
Comment (houseme) [reply apiserver/xdsserverv3/server.go:950]: 已修复

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**